app/controllers/sessions_controller.rb

```diff
@@ -6,11 +6,19 @@ def new
   def create
     user = User.find_by(email: params[:email])
     login(user)
-    
+
     session[:user_id] = user.id
     flash[:success] = "Welcome, #{user.name}, you are logged in!"
   end
 
+  def destroy
+    User.find(session[:user_id]).destroy
+    session[:user_id] = nil
+    session.delete(:cart)
```

Comment:
I think all we need is the session.delete (without the :cart), and the flash and redirect beneath that.